File: scratch/check_braces.js

const fs = require('fs');
const content = fs.readFileSync('src/app/admin/dashboard/page.tsx', 'utf8');
const lines = content.split('\n');

let openBraces = 0;
let inString = false;
let stringChar = '';

for (let i = 0; i < 1754; i++) {
    const line = lines[i];
    for (let j = 0; j < line.length; j++) {
        const char = line[j];
        if (inString) {
            if (char === stringChar && line[j-1] !== '\\') {
                inString = false;
            }
        } else {
            if (char === '"' || char === "'" || char === '`') {
                inString = true;
                stringChar = char;
            } else if (char === '{') {
                openBraces++;
            } else if (char === '}') {
                openBraces--;
            }
        }
    }
}

console.log('Open braces at line 1754:', openBraces);

File: scratch/check_braces.py

with open(r'c:\Users\User\Downloads\ZipSMA\src\app\admin\dashboard\page.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

brace_count = 0
for i, char in enumerate(content):
    if char == '{':
        brace_count += 1
    elif char == '}':
        brace_count -= 1
    
    # Optional: check line numbers if it goes negative
    if brace_count < 0:
        line_num = content[:i].count('\n') + 1
        print(f"Brace count went negative at line {line_num}")
        # Reset to continue checking for other errors
        brace_count = 0

print(f"Final brace count: {brace_count}")
